### ISSUE TITLE
A suggestion for type of ...Response

### ISSUE THREAD

**Issue by SKTimTam** [darseen/amadeus-ts#3]
Hi Darseen,

Checked this repo out for my project and i have to say good job on all the typing, whilst going through the types. I've noticed that this pattern seems to be prominent. Where a ...SearchReturnedResponse is always met with 

Promise<ReturnedResponseSuccess<...SearchGetResult, ...SearchGetResult["data"]>>

Perhaps it could be better for users to actually just get the whole result from the ..SearchReturnedResponse and then .data into the result.
Since  ...SearchGetResult include data that could be useful. i.e dictionaries, meta and etc

tl;dr maybe its better to change the pattern Promise<ReturnedResponseSuccess<...SearchGetResult, ...SearchGetResult["data"]>> **TO** Promise<ReturnedResponseSuccess<...SearchGetResult, ...SearchGetResult>

Happy to hear thoughts, please let me know if i misunderstood something, would not be surprised as it is my first issue raised in Github hahaha thank you :)

**Comment by darseen** [comment]
Hi @SKTimTam, Thank you for taking a look at the project, and raising questions to improve the quality of this library.

It appears that you misunderstood the purpose of "...SearchGetResult["data"]". 
The current structure of the returned response is defined by the Amadeus API, where you get back the `result` property which contains `meta`, `dictionaries`, `warnings` and `data`. And you get the `data` property in the response as well, so instead of getting the data by typing `response.result.data` you just type `response.data`, and both are the same.
You can access `meta`, `dictionaries` and `warnings` through `response.result.meta` etc.

Here is an example with detailed explanation:
```ts
const response = await amadeus.shopping.flightOffersSearch.get({
      originLocationCode: "SYD",
      destinationLocationCode: "BKK",
      departureDate: "2024-12-01",
      adults: 2,
    });
    
// now `response` is of type `ReturnedResponseSuccess<FlightOffersSearchGetResult, FlightOffer[]>`
// `FlightOffersSearchGetResult` is the type of `response.result`
// and `FlightOffersSearchGetResult["data"]` which is `FlightOffer[]` is the type of `response.result.data` or `response.data` 
// ReturnedResponseSuccess<T, K> this is just a generic type that defines the `Response` class that builds the response.
// where `T` is type of `result` and `K` is type of `data`. 
    
// this is the type of `response.result`
type FlightOffersSearchGetResult = {
  meta?: CollectionMetaLink;
  warnings?: Issue[];
  data: FlightOffer[]; // and this is the type for `response.result.data` or `response.data`
  dictionaries?: Dictionaries;
};
```

I know this might be a little confusing, but this is how the API is structured so I had to build the types this way. I really appreciate your question, and hope this helps you better understand this library. Please feel free to ask anything and I will be happy to help.

**Comment by SKTimTam** [comment]
Ah i see, thank you for such a quick response, great explanation. I understnad now.
